core/infra/crypto/signature.cpp

```diff
@@ -58,58 +43,56 @@ std::string sign(const std::string &message,
-  std::vector<unsigned char> pub(pub_decoded.begin(), pub_decoded.end());
-  std::vector<unsigned char> pri(pri_decoded.begin(), pri_decoded.end());
-  return base64::encode(sign(message, pub, pri));
+  return base64::encode(sign(message, pub_decoded, pri_decoded));
+  // byte_array_t pub(pub_decoded.begin(), pub_decoded.end());
+  // byte_array_t pri(pri_decoded.begin(), pri_decoded.end());
+  // return base64::encode(sign(message, pub, pri));
 }
 
 
-std::vector<unsigned char> sign(const std::string &message,
-                                const std::vector<unsigned char> &publicKey,
-                                const std::vector<unsigned char> &privateKey) {
-  std::vector<unsigned char> signature(SIG_SIZE);
+byte_array_t sign(const std::string &message,
+                  const byte_array_t &publicKey,
+                  const byte_array_t &privateKey) {
+  byte_array_t signature(SIG_SIZE);
   ed25519_sign(signature.data(),
-             reinterpret_cast<const unsigned char*>(message.c_str()),
-             message.size(),
-             publicKey.data(),
-             privateKey.data());
+               reinterpret_cast<const byte_t*>(message.c_str()),
+               message.size(),
+               publicKey.data(),
+               privateKey.data());
   return signature;
 }
 
 bool verify(const std::string &signature_b64,
             const std::string &message,
             const std::string &publicKey_b64) {
   return ed25519_verify(base64::decode(signature_b64).data(),
-                        reinterpret_cast<const unsigned char*>(message.c_str()),
+                        reinterpret_cast<const byte_t*>(message.c_str()),
                         message.size(),
                         base64::decode(publicKey_b64).data());
 }
 
-bool verify(const std::vector<unsigned char> &signature,
+bool verify(const byte_array_t &signature,
             const std::string &message,
-            const std::vector<unsigned char> &publicKey) {
+            const byte_array_t &publicKey) {
   return ed25519_verify(signature.data(),
-                        reinterpret_cast<const unsigned char*>(message.c_str()),
+                        reinterpret_cast<const byte_t*>(message.c_str()),
                         message.size(),
                         publicKey.data());
 }
 
 KeyPair generateKeyPair() {
-  std::unique_ptr<unsigned char[]> publicKeyRaw(new unsigned char[sizeof(unsigned char)*32]);
-  std::unique_ptr<unsigned char[]> privateKeyRaw(new unsigned char[sizeof(unsigned char)*64]);
-  std::unique_ptr<unsigned char[]> seed(new unsigned char[sizeof(unsigned char)*32]);
-
-  ed25519_create_seed(seed.get());
-  ed25519_create_keypair(
-    publicKeyRaw.get(),
-    privateKeyRaw.get(),
-    seed.get()
-  );
-
-  return KeyPair(
-     pointer2Vector(std::move(publicKeyRaw), 32),
-     pointer2Vector(std::move(privateKeyRaw), 64)
-   );
+  byte_array_t pub(PUB_KEY_SIZE);
+  byte_array_t pri(PRI_KEY_SIZE);
+  byte_array_t seed(SIG_SIZE);
```

Comment:
it is not `SIG_SIZE` (which is signature size).
It is `seed size`, it never changes and should be `= 32`